src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc

```diff
@@ -1114,29 +1114,27 @@ void grpc_ares_cleanup(void) {}
  */
 
 typedef struct grpc_resolve_address_ares_request {
-  /** the pointer to receive the resolved addresses */
-  grpc_resolved_addresses** addrs_out;
-  /** currently resolving addresses */
-  std::unique_ptr<ServerAddressList> addresses;
-  /** closure to call when the resolve_address_ares request completes */
-  grpc_closure* on_resolve_address_done;
-  /** a closure wrapping on_resolve_address_done, which should be invoked when
-     the grpc_dns_lookup_ares operation is done. */
-  grpc_closure on_dns_lookup_done;
-  /* target name */
-  const char* name;
-  /* default port to use if none is specified */
-  const char* default_port;
-  /* pollset_set to be driven by */
-  grpc_pollset_set* interested_parties;
-  /* underlying ares_request that the query is performed on */
-  grpc_ares_request* ares_request = nullptr;
+  ~grpc_resolve_address_ares_request() { delete ares_request; }
+
+  // synchronizers access to this object (but not to the ares_request itself)
+  absl::Mutex mu;
+  // the pointer to receive the resolved addresses
+  grpc_resolved_addresses** addrs_out ABSL_GUARDED_BY(mu);
+  // currently resolving addresses
+  std::unique_ptr<ServerAddressList> addresses ABSL_GUARDED_BY(mu);
+  // closure to call when the resolve_address_ares request completes
+  grpc_closure* on_resolve_address_done ABSL_GUARDED_BY(mu);
+  // a closure wrapping on_resolve_address_done, which should be invoked when
+  // the grpc_dns_lookup_ares operation is done.
+  grpc_closure on_dns_lookup_done ABSL_GUARDED_BY(mu);
+  // underlying ares_request that the query is performed on
+  grpc_ares_request* ares_request ABSL_GUARDED_BY(mu) = nullptr;
```

Comment:
Suggest making this `std::unique_ptr<>`.

Reply:
done